main/lsp/requests/prepare_rename.cc

```diff
@@ -79,7 +89,12 @@ unique_ptr<ResponseMessage> PrepareRenameTask::runRequest(LSPTypecheckerDelegate
                 if (defResp->symbol.data(gs)->isClassOrModule()) {
                     response->result = getPrepareRenameResult(gs, defResp->symbol);
                 } else if (defResp->symbol.data(gs)->isMethod()) {
-                    response->result = getPrepareRenameResult(gs, defResp->symbol);
+                    if (PrepareRenameTask::validateDef(defResp->termLoc.source(gs))) {
+                        response->result = getPrepareRenameResult(gs, defResp->symbol);
+                    } else {
+                        response->error = make_unique<ResponseError>((int)LSPErrorCodes::InvalidRequest,
+                                                                     "Unsupported method rename operation");
```

Comment:
"Rename Symbol" appears in the menu in any case. But when prepare_rename fails, we get a little "This element can't be renamed" pop up, not the error message :(  Screenshot below.  When rename fails, it's even worse, we get a "No result" pop up at the bottom right. But I think this is probably because we're returning JSONNullObject instead of a proper error message -- I'll change that to send a better error.

If we allow prepare_rename to go through the user will actually have to enter a new name and hit enter to get the error. OTOH we'll be able to give them a better explanation. 🤔 Thoughts?

<img width="483" alt="Screen Shot 2020-11-24 at 2 45 04 PM" src="https://user-images.githubusercontent.com/72465582/100160001-ab915300-2e63-11eb-8263-c910c2c40b62.png">

Reply:
>If we allow prepare_rename to go through the user will actually have to enter a new name and hit enter to get the error. OTOH we'll be able to give them a better explanation. 🤔 Thoughts?

The better explanation sounds like a better UX to me. It confirms to the user that Sorbet understood their request, and cannot support it at the moment. It will also save us debugging time when users encounter the limitation themselves; it's self-documenting.